scripts/ci/run_cli_tool.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+set -euo pipefail
+
+if [ -z "${AIRFLOW_CI_IMAGE}" ]; then
+    echo "Missing environment variable AIRFLOW_CI_IMAGE"
+    exit 1
+fi
+
+SCRIPT_NAME="$( basename "${BASH_SOURCE[0]}")"
+# Drop "-update" suffix, if exists
+TOOL_NAME="$(echo "${SCRIPT_NAME}" | cut -d "-" -f 1)"
+
+SUPPORTED_TOOL_NAMES=("aws" "az" "gcloud" "bq" "gsutil" "terraform" "java")
+
+if [ "$SCRIPT_NAME" == "run_cli_tool.sh" ]
+then
+    # Direct execution - return installation script
+    SCRIPT_PATH=$(readlink -e "${BASH_SOURCE[0]}")
+    >&2 echo "# CLI tool wrappers"
+    >&2 echo "#"
+    >&2 echo "# To install, run the following command:"
+    >&2 echo "#     source <(bash ${SCRIPT_PATH@Q})"
+    >&2 echo "#"
+    >&2 echo ""
+    # Print installation script
+    for NAME in "${SUPPORTED_TOOL_NAMES[@]}"
+    do
+        echo "ln -s ${SCRIPT_PATH@Q} /usr/bin/${NAME}"
+        echo "ln -s ${SCRIPT_PATH@Q} /usr/bin/${NAME}-update"
+        echo "chmod +x /usr/bin/${NAME} /usr/bin/${NAME}-update"
+    done
+    exit 0
+fi
+
+COMMON_DOCKER_ARGS=(
```

Comment:
```suggestion
COTAINER_ID="$(head -n 1 < /proc/self/cgroup | cut -d ":" -f 3 | cut -d "/" -f 3)"

COMMON_DOCKER_ARGS=(
    --ipc "container:${COTAINER_ID}"
    --pid "container:${COTAINER_ID}"
    --network "container:${COTAINER_ID}"
```
@feluelle @potiuk What do you think about it? The Dataflow tasks start the java container. I would like to be able to see if this process was actually started.

Reply:
Why do you think this is necessary? Can we not go just for the exit code?

Reply:
Should we care / monitor the processes inside?

Reply:
In my opinion this sounds more like kind of a debugging functionality rather than functionality we need to kick off those cli's.

Reply:
I run a long-running process using Airflow and I just want to be able to do "ps -aux" to see what process it is The exit-code is not enough, because sometimes the container hangs or runs an interactive terminal. In this situation, I don't have any logs, because the process is waiting for actions from the user.

Reply:
+0. What about you Jarek?

Reply:
Real example:
I working system tests for DataflowCreateJavaJobOperator operator. This operator starts the java process to submit Dataflow job,  This operation may take a while, because it must send the necessary files to Google. Then it waits for the Job ID to appear in the log. If I have problems with the process, I don't have a simple tool to see the whole system situation. I need to check individual containers to check situations. Previously, I could run the `ps -aux` command and see that I have` pytest`, `local task job`,` dataflow`, and others running.  I don't think anyone needs this insulation here.  For a developer, this makes work difficult. For a production environment, this is an important safeguard.

Reply:
I see. This then makes totally sense.

Reply:
Side note: rename `COTAINER_ID` to `CONTAINER_ID`

Reply:
Ye. COTAINER_ID shoudl be renamed. And yes - using the same namespace, ipc and network makes sense - this way we are even closer to run those tools  like if they were installed.

It would be nice to add comment about why we are doing it though - those are very rarely used switches